Fix reference link

GriffinSego opened the pull request:
Fix reference link for LFS api (GitHub Pages 404s for some reason, I replaced link with an archive)

GriffinSego committed:
Add wayback machine link to fix reference

GriffinSego committed:
Fix syntax

britzl reviewed (changes requested):
It seems like the new home for LuaFileSystem and several other Lua modules are: https://lunarmodules.github.io/luafilesystem/index.html

Could you please update the link?

GriffinSego committed:
Update link to new source

Change link from internet archive of old link to new link (https://lunarmodules.github.io/luafilesystem/manual.html#reference)

GriffinSego commented:
I think I've fixed it (this is my first change in a PR), can you confirm? @britzl

britzl reviewed (approved):
Perfect! Thank you.
